DemoApp-Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/RewardedAdActivity.java

```diff
@@ -13,12 +19,81 @@
  */
 public class RewardedAdActivity
         extends AppCompatActivity
+        implements MaxRewardedAdListener
 {
+    private MaxRewardedAd rewardedAd;
 
     @Override
     protected void onCreate(Bundle savedInstanceState)
     {
         super.onCreate( savedInstanceState );
         setContentView( R.layout.activity_rewarded_ad );
+
+        rewardedAd = MaxRewardedAd.getInstance( "YOUR_AD_UNIT_ID", this );
+        rewardedAd.setListener( this );
+
+        // Load the first ad.
+        rewardedAd.loadAd();
+    }
+
+    public void onShowAdClicked(View view)
+    {
+        if ( rewardedAd.isReady() )
+        {
+            rewardedAd.showAd();
+        }
     }
+
+    //region MAX Ad Listener
+
+    @Override
+    public void onRewardedVideoStarted(final MaxAd ad) { }
```

Comment:
Can we follow the sequence we have in the docs? `onRewardedVideoStarted` ,`onRewardedVideoCompleted`, `onUserRewarded`